web/src/selectors/TestDefinition.selectors.ts

```diff
@@ -59,10 +59,17 @@ const TestDefinitionSelectors = () => ({
   selectDefinitionBySelector: createSelector(selectDefinitionList, selectorSelector, (definitionList, selector) =>
     definitionList.find(def => def.selector === selector)
   ),
-  selectAffectedSpans: createSelector(stateSelector, ({affectedSpans}) => affectedSpans),
+  selectAffectedSpans: createSelector(stateSelector, ({affectedSpans, assertionResults, selectedAssertion}) => {
+    const foundAssertion = assertionResults?.resultList.find(({selector}) => selector === selectedAssertion);
```

Comment:
We are checking if the selected assertion exists first

Reply:
This adds the validation at the selector level but the state still has the `affectedSpans` and `selectedAssertion`. I'm wondering if we should clear the state in the corresponding event (revert assertion). 
I'm ok with this approach for now.

Reply:
Yes, I think it would be more complex to run validations against the different CRUD operations and decide when to drop the `selectedAssertion`